src/backend/cuda/cudnn.hpp

```diff
@@ -0,0 +1,44 @@
+/*******************************************************
+ * Copyright (c) 2018, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+#include <cudnn.h>
+#include <common/defines.hpp>
+#include <common/NNHandle.hpp>
+
+namespace cuda
+{
+typedef cudnnHandle_t NNHandle;
```

Comment:
NNHandle, isn't it little too unclear what it is. Probably expand it ? or `NeuralNetHandle`.